Framework/Algorithms/src/SampleCorrections/MCInteractionVolume.cpp

```diff
@@ -46,6 +47,10 @@ MCInteractionVolume::MCInteractionVolume(
   } catch (std::runtime_error &) {
     // swallow this as no defined environment from getEnvironment
   }
+
+  if (m_env) {
```

Comment:
Above the try catch block assigns `m_env` and access through the pointer, so either this check needs to go above (if sample.getEnvironment can return null) or is redundant

My hunch is that we should check for a null ptr before calling `nelements` as we could segfault if it's null

Reply:
This code wasn't changed in this PR, however have made a small edit to make this a bit less obscure.
samle.getEnvironment raises an exception if there isn't an environment. So it can't ever return null. I've added a check for m_env not being null before calling nelements though it doesn't look strange + future proofed in case getEnvironment ever changes to permit null to be returned

Reply:
In that case could we change it to `assert(m_env)` then we guarantee the API stability (which I completely agree with) without having to actually do the check in the wild (since it's redundant)

Reply:
OK done